### ISSUE TITLE
Installing composer with SSH on host for a symfony 3 project

### ISSUE THREAD

**Issue by DvDty** [composer/composer#6509]
My `composer.json`:

```
{
    "name": "/.checkout",
    "license": "proprietary",
    "type": "project",
    "autoload": {
        "psr-4": {
            "": "src/"
        },
        "classmap": [
            "app/AppKernel.php",
            "app/AppCache.php"
        ]
    },
    "autoload-dev": {
        "psr-4": {
            "Tests\\": "tests/"
        },
        "files": [
            "vendor/symfony/symfony/src/Symfony/Component/VarDumper/Resources/functions/dump.php"
        ]
    },
    "require": {
        "php": ">=5.5.9",
        "doctrine/doctrine-bundle": "^1.6",
        "doctrine/doctrine-cache-bundle": "^1.2",
        "doctrine/orm": "^2.5",
        "excelwebzone/recaptcha-bundle": "^1.4",
        "incenteev/composer-parameter-handler": "^2.0",
        "knplabs/knp-paginator-bundle": "^2.6",
        "sensio/distribution-bundle": "^5.0",
        "sensio/framework-extra-bundle": "^3.0.2",
        "symfony/monolog-bundle": "^3.0.2",
        "symfony/polyfill-apcu": "^1.0",
        "symfony/swiftmailer-bundle": "^2.3.10",
        "symfony/symfony": "3.3.*",
        "twig/twig": "^1.0||^2.0"
    },
    "require-dev": {
        "sensio/generator-bundle": "^3.0",
        "symfony/phpunit-bridge": "^3.0"
    },
    "scripts": {
        "symfony-scripts": [
            "Incenteev\\ParameterHandler\\ScriptHandler::buildParameters",
            "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::buildBootstrap",
            "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::clearCache",
            "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::installAssets",
            "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::installRequirementsFile",
            "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::prepareDeploymentTarget"
        ],
        "post-install-cmd": [
            "@symfony-scripts"
        ],
        "post-update-cmd": [
            "@symfony-scripts"
        ]
    },
    "config": {
        "sort-packages": true
    },
    "extra": {
        "symfony-app-dir": "app",
        "symfony-bin-dir": "bin",
        "symfony-var-dir": "var",
        "symfony-web-dir": "web",
        "symfony-tests-dir": "tests",
        "symfony-assets-install": "relative",
        "incenteev-parameters": {
            "file": "app/config/parameters.yml"
        },
        "branch-alias": null
    }
}
```

Output of `composer diagnose`:

```
Loading composer repositories with package information
Updating dependencies (including require-dev)

Fatal error: Out of memory (allocated 840957952) (tried to allocate 67108864 bytes) in phar:///opt/cpanel/composer/bin/composer/src/Composer/DependencyResolver/RuleSetGenerator.php on line 126
zend_mm_heap corrupted
```

When I run this command:

```
composer install
```

Trying to install composer on my hosting, for a symfony 3.3 project.

**Comment by stof** [comment]
`zend_mm_heap corrupted` indicates a bug in PHP itself (or one of the PHP extensions you use). 

regarding the out of memory errors, you should not perform an update on a prod server. This requires lots of memory. Running a `composer install` without having a lock file actually switches to `composer update` (which is why the output says `Updating dependencies`). You should run the update on your development machine and commit the lock file in your project (and then of course include it in what you deploy to your server)

**Comment by DvDty** [comment]
Sorry, for the dumb questions, but I didn't understood what should I do now. I didn't mentioned that I did not upload my vendor folder and the lock file to my project. Should I do that, if it's take that much memory when I'm doing it with composer install? ELI5 me if you could, thank you a lot!

**Comment by stof** [comment]
You don't need to upload your vendor folder (as you are running composer). But you *must* upload your lock file, so that composer can run an install rather than an update. This has several advantages:

- it ensures the prod install the same dependencies than the one you tested locally (running an update would allow it to install versions you never tried, which is not that far from developing directly in prod)
- a composer install takes way less memory than an update (I mean between 10x less to 100x less memory depending on your composer.json)

so yeah, the lock file must be part of the source code being deployed (it must also be part of the source code being checked in your git repo, so that all developers have the same lock file)
-

**Comment by DvDty** [comment]
I did it and I'm getting the exact same error. Also I have this statistic in my cPanel white running the composer update command: 
![image](https://user-images.githubusercontent.com/18031220/27598269-9f369b28-5b6d-11e7-92d3-62e923ddb516.png)
The first one is proccessor usage, the second is memory. This is the error:  
```
composer update 
Loading composer repositories with package information
Updating dependencies (including require-dev)

Fatal error: Out of memory (allocated 840695808) (tried to allocate 67108864 bytes) in phar:///opt/cpanel/composer/bin/composer/src/Composer/DependencyResolver/RuleSetGenerator.php on line 126
zend_mm_heap corrupted
```
